### DISCUSSION TITLE
Wyoming server

### DISCUSSION THREAD

**Discussion by pimdo** [ggml-org/whisper.cpp#1706] [Ideas]
Hi all,
I am using `rhasspy/wyoming-whisper` (docker compose on MacBook Pro M3)  with home-assistant. It works fine, but slow.
I was wondering if it is possible to create a server supporting [wyoming](https://github.com/rhasspy/rhasspy3/blob/master/docs/wyoming.md) protocol.

This is the python [handler](https://github.com/rhasspy/wyoming-faster-whisper/blob/master/wyoming_faster_whisper/handler.py#L17)

Alternatively, it is possible to create a python binding/package for `whisper.cpp`? So the above handler can be updated to interact with `whisper.cpp` (instead of [this](https://github.com/rhasspy/wyoming-faster-whisper/blob/master/wyoming_faster_whisper/faster_whisper/transcribe.py#L80))

**Comment by ser** [comment]
Sure it's possible, but someone needs to do it :)

**Comment by QuantiusBenignus** [comment]
In response to a [feature request](https://github.com/QuantiusBenignus/Blurt/issues/4), I added network transcription support (using the whisper.cpp server implementation) to [Blurt](https://github.com/QuantiusBenignus/blurt) (GNOME extension) and [BlahST](https://github.com/QuantiusBenignus/BlahST) speech-to-text input tools (based on whisper.cpp). I am blown away by the more-than expected speedup of transcription when going to the server!

Before, I was getting ~30x-faster-than-realtime transcription with a local whisper.cpp instance that was loading the model file on each call.
Take a look at the transcription speed when a call to a local whisper.cpp server instance is made (excluding the time for speech input, the curl call takes the bulk of the time in my tools):
![Screenshot from 2024-02-22 11-55-52](https://github.com/ggerganov/whisper.cpp/assets/120202899/cb14ec20-d29d-41cd-9adf-c62d265154c4)

And the request itself (timed to stderr with curl itself)

![Screenshot from 2024-02-22 11-58-06](https://github.com/ggerganov/whisper.cpp/assets/120202899/13d0363d-59f8-4f80-b2ae-43759778a7d9)

This is almost **90x-faster-than-real-time** (~140 ms for a 12.5s speech clip). Loading the model takes about 110 ms for the "main" executable. Seems like there is extra  advantage to running a local server with the model preloaded??? Any thoughts?
(BTW Python requests will be slower than using command line tools)

**Comment by ser** [comment]
So I found someone to write it, it was very myself!

https://github.com/ser/wyoming-whisper-api-client

**Comment by raivisdejus** [comment]
There is also this https://github.com/rhasspy/wyoming-whisper-cpp
